go/genkit/reflection.go

```diff
@@ -334,35 +420,127 @@ func handleRunAction(g *Genkit) func(w http.ResponseWriter, r *http.Request) err
 			json.Unmarshal(body.Context, &contextMap)
 		}
 
-		resp, err := runAction(ctx, g, body.Key, body.Input, body.TelemetryLabels, cb, contextMap)
+		// Run the action with telemetry callback
+		resp, err := runAction(actionCtx, g, body.Key, body.Input, body.TelemetryLabels, cb, contextMap, telemetryCb)
+
+		// Clean up active action if we have a trace ID
+		if traceID != "" {
```

Comment:
```suggestion
		if callbackTraceID != "" {
```

Was a bit hard to follow at a glance, I think this naming might clear it up a bit.

Reply:
Although now that I think about it, is there a reason why we shouldn't just be going off only `resp.Telemetry.TraceID` at this point? It should be filled even in the error case.